src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -429,4 +431,17 @@ private boolean filterDatasetOnProperties(DatasetDTO dataset, String term) {
     private boolean userHasRole(String roleName, Integer dacUserId) {
         return userRoleDAO.findRoleByNameAndUser(roleName, dacUserId) != null;
     }
+
+    public List<Dataset> findAllDatasetsByUser(User user) {
+        List<Dataset> datasets = new ArrayList<>();
+        if (user.hasUserRole(UserRoles.ADMIN)) {
+            datasets.addAll(datasetDAO.getAllDatasets());
+        } else {
+            datasets.addAll(datasetDAO.getActiveDatasets());
+            if (user.hasUserRole(UserRoles.CHAIRPERSON)) {
+                datasets.addAll(datasetDAO.findDatasetsByAuthUserEmail(user.getEmail()));
```

Comment:
If a chairperson has a DAC associated with active datasets, will the datasets appear twice in the returned list (once from `datasetDAO.getActiveDatasets` and a second time from `datasetDAO.findDatasetsByAuthUserEmail`), and if so, would this be an issue?

Reply:
Yes, this is an issue. This is essentially a copy of the original code, but for datasets instead of dataset dtos. The original method collected them in a set which solved this problem. I'll update this, thank you 👍🏽